src/filter-valid-users/index.ts

```diff
@@ -19,5 +19,13 @@ import { FilterValidUsersFn } from './types';
  * ]
  */
 export const filterValidUsers: FilterValidUsersFn = (users) => {
-  throw new Error('Not Implemented');
+  return users.filter((user): user is ValidUser => {
+    return (
+      user.name !== undefined &&
+      user.age !== undefined &&
```

Comment:
If `typeof user.name === 'string'` - it can not be `undefined`.
Too many checks